pkg/restapi/operation/common/models.go

```diff
@@ -102,3 +104,74 @@ type RegisteredServicesResponse struct {
 	// in: body
 	Names []string `json:"names"`
 }
+
+// SendNewMessageRequest model
+//
+// This is used for operation to send new message
+//
+// swagger:parameters sendNewMessage
+type SendNewMessageRequest struct {
+	// Params for sending new message
+	//
+	// in: body
+	Params *SendNewMessageParams
+}
+
+// SendNewMessageParams contains parameters for sending new message
+// with three destination options
+//	1. ConnectionID
+//	2. TheirDID
+//	3. ServiceEndpoint (With recipient Keys, endpoint and optional routing keys)
+type SendNewMessageParams struct {
+
+	// Connection ID of the message destination
+	// This parameter takes precedence over all the other destination parameters.
+	ConnectionID string `json:"connection_ID,omitempty"`
+
+	// DID of the destination.
+	// This parameter takes precedence over `ServiceEndpoint` destination parameter.
+	TheirDID string `json:"their_did,omitempty"`
+
+	// ServiceEndpointDestination service endpoint destination.
+	// This param can be used to send messages outside connection.
```

Comment:
If there is no connection/didexchange, does this protocol internally do didexchange and then send message ?

Reply:
No this is to send message outside existing connection,